src/managers/tidalManager.ts

```diff
@@ -31,11 +32,12 @@ export default class TidalManager {
 				}
 				break;
 			case "playing": {
-				const data = tidalStatus.windowTitle?.trim().split("-");
-				if (!data) return console.error("Can't get current song");
+				let data = tidalStatus.windowTitle?.trim().split(" - ");
+				if (!data)
+					return console.error("Can't get current song");
 
-				const title = data[0].trim(),
-					authors = data[1].trim().split(", ");
+				const title = data[0].trim().substring(0, 40),

```

Comment:
Huh

Reply:
TIDAL's api will sometimes return no results when queried with a very long string. I've moved this truncation to api retry the app already performs

Reply:
examples?

Reply:
Example tracks that return nothing:
https://tidal.com/browse/track/31442419?u
https://tidal.com/browse/track/31442420?u

![image](https://github.com/rxri/tidalRPC/assets/22333584/fe00b9bf-5580-4447-bc56-d368b42963f2)

I'm unsure why this happens, and this is by no means a robust fix, but it successfully finds these tracks

Reply:
umm... okay, i will check the api when i have time so i will come up with better solution